nifi-nar-bundles/nifi-snowflake-bundle/nifi-snowflake-services/src/main/java/org/apache/nifi/snowflake/service/StandardSnowflakeIngestManagerProviderService.java

```diff
@@ -0,0 +1,199 @@
+            .dependsOn(ACCOUNT_IDENTIFIER_FORMAT, AccountIdentifierFormat.ACCOUNT_NAME)
+            .build();
+
+    public static final PropertyDescriptor ACCOUNT_NAME = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(CommonProperties.ACCOUNT_NAME)
+            .dependsOn(ACCOUNT_IDENTIFIER_FORMAT, AccountIdentifierFormat.ACCOUNT_NAME)
+            .build();
+
+    public static final PropertyDescriptor USER_NAME = new PropertyDescriptor.Builder()
+            .name("user-name")
+            .displayName("User Name")
+            .description("The Snowflake user name.")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor PIPE_NAME = new PropertyDescriptor.Builder()
+            .name("pipe-name")
+            .displayName("Pipe Name")
+            .description("The Snowflake pipe's name to ingest from.")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor PRIVATE_KEY_SERVICE = new PropertyDescriptor.Builder()
+            .name("private-key-service")
+            .displayName("Private Key Service")
+            .description("Specifies the Controller Service that will provide the private key. The public key needs to be added to the user account in the Snowflake account beforehand.")
+            .identifiesControllerService(PrivateKeyService.class)
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor HOST_SCHEME = new PropertyDescriptor.Builder()
+            .name("host-scheme")
+            .displayName("Host Scheme")
+            .description("The scheme of the host url to connect to.")
+            .allowableValues("http", "https")
+            .defaultValue("https")
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor HOST_PORT = new PropertyDescriptor.Builder()
+            .name("host-port")
+            .displayName("Host Port")
+            .description("The port of the host url to connect to.")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .defaultValue("443")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(true)
+            .build();
+
+    static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(
+            ACCOUNT_IDENTIFIER_FORMAT,
+            HOST_URL,
+            ACCOUNT_LOCATOR,
+            CLOUD_REGION,
+            CLOUD_TYPE,
+            ORGANIZATION_NAME,
+            ACCOUNT_NAME,
+            USER_NAME,
+            PIPE_NAME,
+            PRIVATE_KEY_SERVICE,
```

Comment:
Please move `Private Key Service` up, just after the User Name because it is the credential belonging to the user.